### PR TITLE
UCL-219 Add github pages support

### PR DESCRIPTION
**Pull request by MichalGorskiOnegini**

### CODE REVIEW

File: catalog-info.yaml [hunk 1]
```diff
@@ -24,7 +28,9 @@ metadata:
       title: Storybook - SNAPSHOT
       icon: docs
   annotations:
+    sonarqube.org/project-key: onewelcome_react-lib-components
     github.com/project-slug: onewelcome/react-lib-components
+    github.com/team-slug: onewelcome/frontend
 spec:
   type: other
```
Comment (jashaj) [inline catalog-info.yaml:35]: type: library

File: catalog-info.yaml [hunk 2]
```diff
@@ -24,7 +28,9 @@ metadata:
...
   lifecycle: unknown
```
Comment (jashaj) [inline catalog-info.yaml:36]: lifecycle: production?

File: catalog-info.yaml [hunk 3]
```diff
@@ -16,6 +16,10 @@ apiVersion: backstage.io/v1alpha1
 kind: Component
 metadata:
   name: react-lib-components
+  description: React component library
+  tags:
+    - typescript
+    - react
   links:
```
Comment (jashaj) [inline catalog-info.yaml:23]: Should the release URL point to GitHub pages?

File: catalog-info.yaml [hunk 4]
```diff
@@ -16,6 +16,10 @@ apiVersion: backstage.io/v1alpha1
 kind: Component
 metadata:
   name: react-lib-components
+  description: React component library
```
Comment (jashaj) [inline catalog-info.yaml:19]: React component library for what? :-)

File: README.md [hunk 5]
```diff
@@ -1,5 +1,12 @@
 # OneWelcome React Component Library
 
+[![OneWelcome](https://circleci.com/gh/onewelcome/react-lib-components.svg?style=svg)](<https://circleci.com>)
```
Comment (jashaj) [inline README.md:3]: The alt text is incorrect here. Should be `CircleCI status`

File: .circleci/config.yml [hunk 6]
```diff
@@ -64,6 +64,18 @@ jobs:
...
             - project/storybook-static
 
+  deploy-storybook-to-github-pages:
+    executor: onegini-build/node-executor
+    steps:
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: Set git repository
+          command: git remote set-url origin https://git:${GITHUB_API_TOKEN}@github.com/onewelcome/react-lib-components.git
```
Comment (jashaj) [inline .circleci/config.yml:74]: Can you verify that it does not print the actual API token in log files?

File: package.json [hunk 7]
```diff
@@ -29,7 +30,8 @@
     "prettier": "prettier --write --ignore-path .gitignore \"**/*.+(ts|tsx|scss|mdx)\"",
     "storybook": "start-storybook -p 6006",
     "test": "dts test --passWithNoTests",
-    "test:watch": "dts test --passWithNoTests --watch"
+    "test:watch": "dts test --passWithNoTests --watch",
+    "github-pages:deploy": "gh-pages -d storybook-static -u \"github-actions-bot <support+actions@github.com>\""
```
Comment (jashaj) [inline package.json:34]: IMO this part of the package.json should move to the CircleCI config. We don't want people to manually deploy docs. Suggestion: https://circleci.com/blog/deploying-documentation-to-github-pages-with-continuous-integration/

File: catalog-info.yaml (hunk 3 above)
Comment (MichalGorskiOnegini) [reply catalog-info.yaml:23]: for now, let's leave it as is. In the near future we might completely move to gh pages with it (or somewhere else)

File: catalog-info.yaml (hunk 2 above)
Comment (pawel-napieracz) [reply catalog-info.yaml:36]: @MichalGorskiOnegini ?

**Review by pawel-napieracz — CHANGES_REQUESTED** [review]

File: catalog-info.yaml (hunk 2 above)
Comment (MichalGorskiOnegini) [reply catalog-info.yaml:36]: I'm not sure so I did not change it

**Review by pawel-napieracz — APPROVED** [review]

**Pull request merged**